bagel/cli.py

```diff
@@ -496,27 +496,31 @@ def bids(
     )
     bids_dataset = file_utils.load_tabular(bids_table, input_type="BIDS")
 
-    if not bids_dataset["suffix"].isin(mappings.BIDS.keys()).any():
-        log_error(
-            logger,
-            f"No Neurobagel-supported BIDS suffixes found in BIDS table 'suffix' column: {bids_table}. "
-            "No imaging metadata could be added to the subject graph data. "
-            "Please ensure your dataset includes at least one image file with a Neurobagel-supported BIDS suffix "
-            f"(supported suffixes: {list(mappings.BIDS.keys())}).",
-        )
+    # NOTE: Even though the BIDS table model validation will check for required columns,
+    # since we want to ignore unsupported suffixes with a warning instead of just a validation error,
+    # we first check the suffix column separately to customize the error/warning behaviour.
```

Comment:
I like the NOTE, but I don't get it. 

Is it:

- we don't check the suffixes against all BIDS suffixes with pydantic because that would make an error
- instead we have our own custom suffix-checker that only warns
- but that requires a "suffix" column to exist, otherwise KeyError
?

Or will pydantic check the suffixes again later anyway?

Reply:
i.e.: why can't we run the pydantic validation first to ensure all columns exist  if it ignores the suffixes? And if it does not ignore the suffixes (i.e. validates them again later), then why does the CLI not error out at that point anyway?

Reply:
Yes to all 3 of your above points. The Pydantic model on its own would still check the values of the `suffix` column against our supported BIDS suffixes, but it's not smart enough to do anything other than emit a validation error when it finds an unsupported suffix. By contrast, we want to 1) warn the user instead 2) filter out the unrecognized suffixes and proceed using the filtered BIDS table. So, instead of just validating against the Pydantic model upfront, we do the checking, warning, and filtering of the `suffix` column 'manually', and then 'verify' that the final filtered table is compliant with our table model (via the Pydantic model validation, which will also check the other columns as well which do not require special handling).